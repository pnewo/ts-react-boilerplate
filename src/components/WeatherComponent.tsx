import * as React from 'react';

export interface IWeatherComponentProps {
    readonly weather?: string;
    getWeather(lat: number, lon: number): void;
}

export default class WeatherComponent extends React.Component<IWeatherComponentProps, undefined> {
    componentDidMount() {
        const { weather, getWeather } = this.props;
        if (!weather) {
            if ('geolocation' in navigator) {
                navigator.geolocation.getCurrentPosition(position =>
                    getWeather(position.coords.latitude, position.coords.longitude));
            } else {
                // Client didn't give permission to use geolocation, using Helsinki coordinates
                const helLat = 60.1699;
                const helLon = 24.9384;
                getWeather(helLat, helLon);
            }
        }
    }

    render() {
        return (<div className="current-weather">Current weather: {this.props.weather}</div>);
    }
}
