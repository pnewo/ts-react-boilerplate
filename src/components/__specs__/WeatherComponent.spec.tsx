import * as React from 'react';
import { mount } from 'enzyme';
import WeatherComponent from '../WeatherComponent';

describe('WeatherComponent', () => {
    const latitude = 60.1699;
    const longitude = 24.9384;
    const weather = 'cloudy';
    const getWeather = jest.fn();
    const notCalled = jest.fn();
    const wrapperCall = mount(<WeatherComponent getWeather={getWeather} />);
    const wrapperNoCall = mount(<WeatherComponent weather={weather} getWeather={notCalled} />);

    it('should render correctly', () => {
        expect(wrapperCall).toMatchSnapshot();
        expect(wrapperNoCall).toMatchSnapshot();
    });

    it('should have called getWeather on mount', () => {
        expect(getWeather).toBeCalledWith(latitude, longitude);
    });

    it('should not have called getWeather', () => {
        expect(notCalled).toHaveBeenCalledTimes(0);
    });
});
