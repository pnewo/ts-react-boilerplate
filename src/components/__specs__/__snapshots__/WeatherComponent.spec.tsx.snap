// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherComponent should render correctly 1`] = `
<WeatherComponent
  getWeather={[Function]}
>
  <div
    className="current-weather"
  >
    Current weather: 
  </div>
</WeatherComponent>
`;

exports[`WeatherComponent should render correctly 2`] = `
<WeatherComponent
  getWeather={[Function]}
  weather="cloudy"
>
  <div
    className="current-weather"
  >
    Current weather: 
    cloudy
  </div>
</WeatherComponent>
`;
